Subscription refactoring #5

federicocappelli opened the pull request:
Task/Issue URL: https://app.asana.com/0/1205842942115003/1206805455884775/f
Tech Design URL: https://app.asana.com/0/1205842942115003/1207147511614062/f
CC: @miasma13 

**Description**:

This PR updates BSK from https://github.com/duckduckgo/BrowserServicesKit/pull/874 and contains all the needed refactoring for the Subscription classes inits.
It also contains a new unit test class:
- SubscriptionPagesUseSubscriptionFeatureTests

Complete tests will be implemented in follow-up tasks

**Steps to test this PR**:
Subscription must work as usual, manual smoke test script available in this [README](https://github.com/duckduckgo/BrowserServicesKit/blob/main/Sources/SubscriptionTestingUtilities/README.md)

**Definition of Done (Internal Only)**:

* [x] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Device Testing**:

* [x] iPhone 15 Pro

**OS Testing**:

* [x] iOS 17
* [ ] 
---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

federicocappelli committed:
subscription refactoring for tests

federicocappelli committed:
Merge branch 'main' into fcappelli/subscription_refactoring_5

# Conflicts:
#	DuckDuckGo.xcodeproj/project.pbxproj
#	DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved

federicocappelli committed:
lint

federicocappelli committed:
unit tests fixed

afterxleep commented on DuckDuckGo/Subscription/SubscriptionManageriOS14.swift:
Hopefully you can get rid of this one now that Sam merged the iOS 14 drop?

federicocappelli replied:
YES! I already have a stash with all changes, I'm just waiting for Sam's modifications to be live, I would like to avoid a cherry pick if something goes wrong with the iOS 14 drop

afterxleep commented on DuckDuckGoTests/Subscription/SubscriptionPagesUseSubscriptionFeatureTests.swift:
```diff
@@ -0,0 +1,47 @@
+final class SubscriptionPagesUseSubscriptionFeatureTests: XCTestCase {
+
+    override func setUpWithError() throws {
+        // Put setup code here. This method is called before the invocation of each test method in the class.
```
Remove these if not used?

afterxleep reviewed (approved):
LGTM

federicocappelli committed:
BSK 165.0.0

Merge branch 'main' into fcappelli/subscription_refactoring_5

# Conflicts:
#	DuckDuckGo.xcodeproj/project.pbxproj
#	DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
